### ISSUE TITLE
Convert to machines API instead of flyctl?

### ISSUE THREAD

**Issue by alexellis** [jsiebens/inlets-on-fly#2]
@mrkurt was telling me about Fly's new HTTP API for machines and apps.

Perhaps instead of exec'ing flyctl, you could make HTTP requests to the API?

Would that fit into the interface for [cloud-provision](https://github.com/inlets/cloud-provision)

```go
type Provisioner interface {
	Provision(BasicHost) (*ProvisionedHost, error)
	Status(id string) (*ProvisionedHost, error)
	Delete(HostDeleteRequest) error
}
```

How cool would it be if inletsctl/inlets-operator both could provision to fly-hosted MicroVMs?

**Comment by jsiebens** [comment]
Yes, I agree, that would be much better. (It would also be a fix for #3) 

If it fits into the interface for cloud-provision, then this tool would not be necessary anymore in favor of inletsctl/inlets-operator
